specutils/manipulation/utils.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import division
+
+import numpy as np
+
+from ..spectra import Spectrum1D, SpectralRegion
+
+__all__ = ['excise_regions']
+
+
+def linear_exciser(spectrum, region):
+    """
+    Basic spectral excise method where the spectral region defined by the
+    2-tuple parameter `region` (start and end wavelengths) will result
+    in the flux between those regions be set to a linear ramp of the
+    two points immediately before and after the start and end regions.
+
+    Other methods could be defined by the user to do other types of excision.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.Spectrum1D`
+        The `~specutils.Spectrum1D` object to which the smoothing will be applied.
+
+    region : `~specutils.SpectralRegion`
+        Region to excise.
+
+    Returns
+    -------
+    spectrum : `~specutils.Spectrum1D`
+        Output `~specutils.Spectrum1D` with the region excised.
+
+    Raises
+    ------
+    ValueError
+       In the case that ``spectrum`` and ``regions`` are not the correct types.
+
+    """
+
+    #
+    # Find the indices of the wavelengths in the range `range`
+    #
+
+    wavelengths = spectrum.spectral_axis
+    wavelengths_in = (wavelengths >= region.lower) & (wavelengths < region.upper)
+    inclusive_indices = np.nonzero(wavelengths_in)[0]
+
+    #
+    # Now set the flux values for these indices to be a
+    # linear range
+    #
+
+    s, e = inclusive_indices[0]-1, inclusive_indices[-1]+1
+
+    flux = spectrum.flux.value
+    modified_flux = flux
+    modified_flux[s:e] = np.linspace(flux[s], flux[e], len(inclusive_indices)+1)
+
+    # Return a new object with the regions excised.
+    return Spectrum1D(flux=modified_flux*spectrum.flux.unit, spectral_axis=wavelengths,
+                      uncertainty=spectrum.uncertainty,
+                      wcs=spectrum.wcs, unit=spectrum.unit,
+                      spectral_axis_unit=spectrum.spectral_axis_unit,
+                      velocity_convention=spectrum.velocity_convention,
+                      rest_value=spectrum.rest_value)
+
+
+def excise_regions(spectrum, regions, exciser=linear_exciser):
```

Comment:
I don't fully understand the motivation for this function.  Why not just implement *only* `linear_exciser` for now, and have future excisers be their own function?

Reply:
It was implemented primarily to get around the multiple region thing.  This way anyone who adds their own excise method (either in here, or passes in their own magical method) does not have to worry about the fact that regions could be a list.  So, all they have to implement is assuming one region.

Reply:
Ohh, gotcha.  I wonder if we should note that this is likely to be replaced by the multiple-regions thing?

Reply:
After thinking about it again I guess this should be decided in #289 - i.e., that might change it, but better to get this in and deal with it there.